code/xcvm/cosmwasm/contracts/gateway/src/error.rs

```diff
@@ -38,4 +38,15 @@ pub enum ContractError {
 	UnknownChannel,
 	#[error("The asset is already registered.")]
 	AlreadyRegistered,
+	#[error("Route not found.")]
+	RouteNotFound,
+	#[error("{0:?}")]
+	Bech32(bech32_no_std::Error),
```

Comment:
```suggestion
	Bech32(#[from] bech32_no_std::Error),
```
and than you won’t need to make the From implementation yourself.